objects/client.c

```diff
@@ -2153,13 +2161,29 @@ client_set_icon_from_pixmaps(client_t *c, xcb_pixmap_t icon, xcb_pixmap_t mask)
 
 /** Kill a client.
  *
+ * By default, this will use the normal X11 way of closing a client (the `X`
+ * button). If a signal number is provided, it will be used instead.
+ *
+ * This can be used to pause or wakeup some clients or to save energy.
+ *
+ * @tparam[opt=nil] integer signal The POSIX signal number.
  * @function kill
  */
 static int
 luaA_client_kill(lua_State *L)
 {
     client_t *c = luaA_checkudata(L, 1, &client_class);
-    client_kill(c);
+    int sig = -1;
+
+    /* Allow a signal number */
```

Comment:
No. Just no!

Feel free to deprecate `c:kill()`, rename it to `c:close()` and add `c:send_signal(number)` (or `c:signal(number)`?), but having the same function do two completely different things seems like a really, really bad idea. I guess that the only reason why you want to do it like this is because it's called `kill`.